.github/workflows/tools.yml

```
@@ -21,12 +21,10 @@ jobs:
             subsystem: tools
             label: tools
             run: |
-              NEW_VERSION=$(npm view eslint dist-tags.latest)
-              CURRENT_VERSION=$(node -p "require('./tools/node_modules/eslint/package.json').version")
-              if [ "$NEW_VERSION" != "$CURRENT_VERSION" ]; then
-                echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
-                ./tools/dep_updaters/update-eslint.sh
-              fi
+              ./tools/dep_updaters/update-eslint.sh > temp-output
+              cat temp-output
+              tail -n1 temp-output | grep "NEW_VERSION=" >> "$GITHUB_ENV"
```

I was under the impression that `grep` would exit with a 0 exit code as long as the `-q` version were not passed, but that doesn't seem to be true, According to `man grep`:

```
EXIT STATUS
     The grep utility exits with one of the following values:

     0     One or more lines were selected.
     1     No lines were selected.
     >1    An error occurred.
```
We would need to catch this, otherwise the workflow would be marked as failed.
```suggestion
              tail -n1 temp-output | grep "NEW_VERSION=" >> "$GITHUB_ENV" || true
```

---

Thanks ^^ I should create a sheet cheat of stuff I should consider when I write a bash/shell script.

---

Working on the ada script is in my mind: https://github.com/nodejs/node/pull/46488#issuecomment-1435687813

WDYT about merging this first? Let one of them pass, then refactor the add and move them.